DEVELOPER.md

```diff
@@ -426,3 +431,56 @@ Tests should ideally follow a few conventions for easier maintenance:
   - `@pytest.mark.regression` if the test compares results from different versions
 
 **Note:** If all three external model repositories are not installed as described above, some tests will be skipped. The full test suite includes >750 cases. All must pass before changes can be merged into this repository.
+
+## Git Strategy for Managing Long-Lived Branches
+
+When a feature branch takes a long time to develop, it is easy to become out of sync with the develop branch.  Depending on the situation, it may be advisable to periodically squash the commits on the feature branch and rebase the change set with develop.  The following approach for updating a long-lived feature branch has proven robust.
+
+In the example below, the feature branch is assumed to be called `feat-xyz`
+
+### Create a Backup
+
+Begin by creating a backup copy of the feature branch in case anything goes terribly wrong.
+
+```
+git checkout feat-xyz
+git checkout feat-xyz-backup
+```
+
+### Squash Feature Branch Commits
+
+Next, consider squashing commits on the feature branch.  If there are many commits, it is beneficial to squash them before trying to rebase with develop.  There is a nice article on [squashing commits into one using git](https://www.internalpointers.com/post/squash-commits-into-one-git), which has been very useful for consolidating commits on a long-lived modflow6 feature branch.
+
+Once the commits on the feature branch have been consolidated, a force push to origin is recommended.  This will update `feat-xyz` on origin.
```

Comment:
I don't think pushing here is strictly necessary, except as an intermediate backup/checkpoint so the squashed branch state can be retrieved if rebasing fails. But there is also `rebase --abort` to unwind a rebase that's gotten out of hand.

A quick and dirty way to squash without interactive rebase is
- `git reset --soft <first commit on the feature branch>`
- `git commit --amend -m "consolidated commit message"`

This is probably ill-advised if no backup of the branch has been made (accidentally typing `--hard` will wipe out all your work).

Reply:
Nice alternative.  Mentioned in revision.